pipelines/upgrade_pipeline.yml

```diff
@@ -106,6 +104,8 @@
     - foreman_server_repositories
     - foreman_client_repositories
     - role: foreman_installer
+      foreman_installer_upgrade: True
+      foreman_installer_options_internal_use_only: []
```

Comment:
why do you reset all internal options?

Reply:
One problem I kept running into was that it's adding the initial install options to proxies on upgrades. Since we don't change options on upgrade (and the actual roles add the upgrade statements), I thought we should clear them out.

Reply:
Feels like the role should ignore them then?

Reply:
Not always. In the future we might want to add testing scenarios where we enable a module that was introduced in 1.next so the role is not the right layer.

Reply:
Btw, another solution I thought of was to change the param to `server_installer_options` and explicitly set that in the initial install. Perhaps we're also getting to a point where a simulated inventory is actually nice. I'm not sure our vagrant inventory plugin exposes the variables, but that might be a better place.

Reply:
The problem with moving these to the inventory is that the inventory is evaluated for the whole playbook once, so it can't change mid-way (there are ways to refresh it, but I'm not a fan).

Reply:
But we can store custom variables. `server_installer_options` is not used elsewhere. In the playbook during the initial run we can do `foreman_installer_options_internal_use_only: "{{ server_installer_options }}"` just where we need it.